### PR TITLE
Add minimal functional test

### PR DESCRIPTION
**Pull request by franmomu**
This is the minimal functional test I can think of. Not sure if it is the way it should be...

### CODE REVIEW

**Comment by core23** [comment]
Why did you tag this as `minor`. This should be `pedantic`, because this is just a test

**Comment by phansys** [comment]
> Why did you tag this as `minor`. This should be `pedantic`, because this is just a test

You're right, my bad.

File: tests/Functional/Controller/CoreControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Functional\Controller;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\AdminBundle\Tests\Functional\AppKernel;
+use Symfony\Bundle\FrameworkBundle\Client;
+
+class CoreControllerTest extends TestCase
```
Comment (phansys) [inline tests/Functional/Controller/CoreControllerTest.php]: Just to increase the code coverage measurement for `DashboardAction`, I'd call this class `DashboardActionTest`.

Comment (franmomu) [reply tests/Functional/Controller/CoreControllerTest.php]: Done

File: tests/Functional/AppKernel.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Bundle\SecurityBundle\SecurityBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\Routing\RouteCollectionBuilder;
+
+class AppKernel extends Kernel
```
Comment (core23) [inline tests/Functional/AppKernel.php]: You can make test classes final

**Comment by franmomu** [comment]
There is still one thing it doesn't work and I don't understand. In [the travis job](https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/605114344#L645) with `COMPOSER_FLAGS="--prefer-lowest"` the error is:
```
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "fragment.renderer.hinclude".
```

I have no idea where this comes from, I tried to reproduce it, but I thought it was going to install `2.8` for Symfony packages, but this is what is actually installing in Travis:
```
composer update --prefer-dist --no-interaction --prefer-stable ${COMPOSER_FLAGS}
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 7 installs, 65 updates, 13 removals
  - Removing symfony/var-exporter (v4.3.5)
  - Removing symfony/translation-contracts (v1.1.7)
  - Removing symfony/service-contracts (v1.1.7)
  - Removing symfony/polyfill-php72 (v1.12.0)
  - Removing symfony/polyfill-intl-idn (v1.12.0)
  - Removing symfony/mime (v4.3.5)
  - Removing symfony/inflector (v4.3.5)
  - Removing symfony/event-dispatcher-contracts (v1.1.7)
  - Removing symfony/contracts (v1.1.0)
  - Removing symfony/cache-contracts (v1.1.7)
  - Removing doctrine/reflection (v1.0.0)
  - Removing doctrine/event-manager (v1.0.0)
  - Removing doctrine/persistence (1.1.1)
  - Downgrading doctrine/inflector (v1.3.0 => v1.2.0): Loading from cache
  - Downgrading symfony/polyfill-mbstring (v1.12.0 => v1.3.0): Loading from cache
  - Downgrading symfony/polyfill-ctype (v1.12.0 => v1.8.0): Loading from cache
  - Installing paragonie/random_compat (v1.0.0): Loading from cache
  - Downgrading psr/cache (1.0.1 => 1.0.0): Loading from cache
  - Downgrading symfony/event-dispatcher (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading psr/log (1.1.1 => 1.0.0): Loading from cache
  - Downgrading symfony/finder (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading symfony/filesystem (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading symfony/debug (v4.3.5 => v2.8.0): Loading from cache
  - Downgrading symfony/class-loader (v3.4.32 => v3.2.0): Loading from cache
  - Installing symfony/polyfill-apcu (v1.1.0): Loading from cache
  - Installing psr/simple-cache (1.0.0): Loading from cache
  - Downgrading symfony/config (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading symfony/cache (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading symfony/http-foundation (v4.3.5 => v3.3.11): Loading from cache
  - Downgrading symfony/http-kernel (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading symfony/routing (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading symfony/dependency-injection (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading symfony/framework-bundle (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading doctrine/cache (v1.8.1 => v1.0): Loading from cache
  - Installing symfony/polyfill-php54 (v1.0.0): Loading from cache
  - Installing symfony/polyfill-php70 (v1.0.0): Loading from cache
  - Installing symfony/polyfill-util (v1.0.0): Loading from cache
  - Installing symfony/polyfill-php56 (v1.0.0): Loading from cache
  - Downgrading symfony/property-access (v4.3.5 => v2.8.0): Loading from cache
  - Downgrading symfony/security (v4.2.11 => v3.2.0): Loading from cache
  - Downgrading symfony/security-acl (v3.0.2 => v3.0.0): Loading from cache
  - Downgrading doctrine/lexer (1.0.2 => v1.0): Loading from cache
  - Downgrading doctrine/annotations (v1.8.0 => v1.0): Loading from cache
  - Downgrading symfony/templating (v4.3.5 => v2.8.0): Loading from cache
  - Downgrading knplabs/knp-menu (2.6.0 => 2.3.0): Loading from cache
  - Downgrading knplabs/knp-menu-bundle (v2.3.0 => v2.2.0): Loading from cache
  - Downgrading twig/twig (v2.12.1 => v2.10.0): Loading from cache
  - Downgrading symfony/twig-bridge (v4.2.11 => v3.4.0): Loading from cache
  - Downgrading symfony/twig-bundle (v4.1.0 => v3.2.0): Loading from cache
  - Downgrading symfony/options-resolver (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading symfony/polyfill-intl-icu (v1.12.0 => v1.0.0): Loading from cache
  - Downgrading symfony/intl (v4.3.5 => v2.8.18): Loading from cache
  - Downgrading symfony/form (v4.2.11 => v3.4.0): Loading from cache
  - Downgrading symfony/console (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading symfony/asset (v4.3.5 => v3.3.0): Loading from cache
  - Downgrading doctrine/collections (v1.6.2 => v1.0): Loading from cache
  - Downgrading doctrine/common (v2.11.0 => v2.7.0): Loading from cache
  - Downgrading doctrine/dbal (v2.9.2 => v2.4.0): Loading from cache
  - Downgrading sonata-project/doctrine-extensions (1.3.1 => 1.1.0): Loading from cache
  - Downgrading twig/extensions (v1.5.4 => v1.5.0): Loading from cache
  - Downgrading symfony/translation (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading symfony/validator (v4.3.5 => v3.4.0): Loading from cache
  - Downgrading sonata-project/datagrid-bundle (2.5.0 => 2.2): Loading from cache
  - Downgrading cocur/slugify (v3.2 => v1.4): Loading from cache
  - Downgrading sonata-project/core-bundle (3.17.0 => 3.15.0): Loading from cache
  - Downgrading sonata-project/cache (2.0.1 => 1.0.0): Loading from cache
  - Downgrading sonata-project/block-bundle (3.18.1 => 3.17.0): Loading from cache
  - Downgrading sonata-project/exporter (1.11.1 => 1.11.0): Loading from cache
  - Downgrading symfony/doctrine-bridge (v4.2.11 => v2.8.0): Loading from cache
  - Downgrading symfony/expression-language (v4.3.5 => v2.8.0): Loading from cache
  - Downgrading symfony/security-bundle (v4.0.15 => v3.2.0): Loading from cache
  - Downgrading jms/metadata (1.7.0 => 1.1.1): Loading from cache
  - Downgrading jms/cg (1.2.0 => 1.1.0): Loading from cache
  - Downgrading jms/aop-bundle (1.3.0 => 1.2.0): Loading from cache
  - Downgrading symfony/process (v3.4.32 => v3.3.0): Loading from cache
  - Downgrading jms/di-extra-bundle (1.9.2 => 1.9.0): Loading from cache
  - Downgrading nikic/php-parser (v4.2.5 => v4.0.0): Loading from cache
  - Downgrading jms/translation-bundle (1.4.4 => 1.4.2): Loading from cache
  - Downgrading symfony/yaml (v3.4.32 => v3.4.0): Loading from cache
  - Downgrading sensio/generator-bundle (v3.1.7 => 3.1.0): Loading from cache
  - Downgrading sonata-project/intl-bundle (2.7.0 => 2.4.0): Loading from cache
  - Removing symfony/dom-crawler (v4.3.5)
  - Installing symfony/dom-crawler (2.0.5): Loading from cache
  - Downgrading symfony/browser-kit (v4.3.5 => v2.8.0): Loading from cache
  - Downgrading symfony/phpunit-bridge (v4.3.5 => v4.3.0): Loading from cache
  - Downgrading matthiasnoback/symfony-config-test (4.0.1 => 4.0.0): Loading from cache
```

**Comment by greg0ire** [comment]
haha some of our deps are probably no longer compatible with 2.8, which means we can probably drop the constraint ourselves… note that's not true for `symfony/browser-kit` though.

**Comment by franmomu** [comment]
😄 it was the `symfony/twig-bundle` that is installing the `3.2` version, so maybe the Symfony packages could also be raised from `3.2` to `3.4`.  Now [there is other one](https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/605373121#L645).

**Comment by franmomu** [comment]
> haha some of our deps are probably no longer compatible with 2.8, which means we can probably drop the constraint ourselves… note that's not true for `symfony/browser-kit` though.

So... is it ok if I make a PR dropping `2.8` Symfony packages and raising to `3.4`? If so I guess `sensio/generator-bundle` can be dropped as well and use `maker-bundle`. Or should this be done in `master` branch?

**Comment by franmomu** [comment]
After some digging... the error looks like this error is fixed using version `1.1.3` of `https://github.com/sonata-project/sonata-doctrine-extensions`.

But then I get:
```
Twig\Error\RuntimeError: The "core" extension is not enabled.
```
[Which apparently is more difficult to deal with](https://github.com/symfony/symfony/issues/20284#issuecomment-255611646). 🤔

**Comment by franmomu** [comment]
It's finally green 😌

**Pull request merged**